crates/ere-jolt/src/lib.rs

```diff
@@ -24,53 +28,73 @@ pub struct JOLT_TARGET;
 impl Compiler for JOLT_TARGET {
     type Error = JoltError;
 
-    type Program = Program;
+    // Note that we use `PathBuf` as `Program` because the only way to pass elf
+    // information to `jolt::host::Program` is by setting elf path, so we don't
+    // bother to read it into memory here.
+    type Program = PathBuf;
```

Comment:
I guess should be fine if the compilation and further use of the program happen close in time (imagining if something could delete that file), and always on the same machine.